hrm-service/src/contact-job/contact-job-complete.ts

```diff
@@ -71,6 +71,11 @@ export const pollAndProcessCompletedContactJobs = async (jobMaxAttempts: number)
   const completedJobs = await Promise.allSettled(
     messages.map(async m => {
       try {
+
+        // Immediately handle the message deletion in case of error since poller
+        // is responsible for retrying failed jobs.
+        await deleteCompletedContactJobsFromQueue(m.ReceiptHandle);
```

Comment:
Burying this was causing issues where jobs would never be deleted from the queue and would be processed repeatedly forever which was probably the root cause of the db overload... some of the failedAttemptPayloads were enormous JSON arrays. That exposed a flaw that would have bitten us eventually, but failure to remove messages from the complete queue was the root cause. 

As soon as the poller picks up a job from the queue it is responsible for handling it so deleting the job from the queue should be safe. Eventually, keeping track of the status of a job in each hop along its lifecycle becomes necessary in my experience to prevent double work, lock for concurrency, and handle other edge cases. But it isn't that big of a deal for these jobs where double work shouldn't hurt anything as long... as long as it isn't an infinite loop.

Reply:
@GPaoloni I am about to merge this, but wanted you to see this change.